tests/MockApPipe.yaml

```diff
@@ -50,16 +51,14 @@ contracts:
   - imageDifference.doSkySources == transformDiaSrcCat.doRemoveSkySources
   # Inputs and outputs must match
   # Use of ConnectionsClass for templated fields is a workaround for DM-30210
-  - retrieveTemplate.connections.ConnectionsClass(config=retrieveTemplate).outputExposure.name ==
-      imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diffIm.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).diaSources.name ==
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).warpedExposure.name ==
-        diaPipe.connections.ConnectionsClass(config=diaPipe).warpedExposure.name
+        diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
```

Comment:
Shouldn't you rename both terms, as you did in `ap_pipe`?

Reply:
No, I need to keep some of the ones in the mock pipeline, because it's based on `ImageDifferenceTask`, not `ImageDifferenceFromTemplateTask`. We'll want to rework these to use the new SubtractTask anyway.